### ISSUE TITLE
DynamoDB: Support list with objects of different types.

### ISSUE THREAD

**Issue by surister** [crate/commons-codec#28]
Example:
```python
[
  {'a': 1}, 2, "Three"
]
```

**Comment by surister** [comment]
That's actually ilegal in CrateDB, but ok in DynamoDB, we should probably detect it and handle it.

**Comment by amotl** [comment]
Yeah, CrateDB needs to specify the inner type, like `::ARRAY(TEXT)`. So, it is not possible to store lists of varied types.

**Comment by amotl** [comment]
> we should probably detect it and handle it

Good idea. While storing it 1:1 is probably not possible, because CrateDB can't handle it, we need a trick/workaround. Do you have any suggestions? Maybe one of you already applied any tricks in similar situations, or have another idea?

/cc @hlcianfagna, @wierdvanderhaar, @hammerhead, @proddata, @matriv

**Comment by surister** [comment]
How about we detect it, map it to a `OBJECT DYNAMIC` and insert it to the object with keys being the types? 
```py
l = [
  {'a': 1}, 2, "Three"
]

r = {}

for el in l:
    if (_type := type(el).__name__) not in r:
        r[_type] = []
    r[_type].append(el)

print(r)
```

```py
{
    'dict': [{'a': 1}],
    'int': [2],
    'str': ['Three']
}
```

**Comment by proddata** [comment]
You can indirectly store it

```sql
cr> CREATE TABLE t01 (o OBJECT(IGNORED));
cr> INSERT INTO t01 (o) VALUES ($${"oo":[{"a": 1}, 2, "Three"]}$$);

cr> SELECT * FROM t01;
+--------------------------------+
| o                              |
+--------------------------------+
| {"oo": [{"a": 1}, 2, "Three"]} |
+--------------------------------+

cr> SELECT o['oo'][1] FROM t01;
+------------+
| o['oo'][1] |
+------------+
| {"a": 1}   |
+------------+

cr> SELECT o['oo'][2] FROM t01;
+------------+
| o['oo'][2] |
+------------+
|          2 |
+------------+
```

**Comment by matriv** [comment]
I think the `OBJECT(IGNORED)` is the only way, as @proddata suggested, but then nothing is indexed, so maybe this functionality can be enabled/disabled with a flag, and it should be clearly documented how such rows are indexed, and that searching on them is not possible.

**Comment by proddata** [comment]
I think the option is to store what is possible in a typed field and what is in a separate column.
Ideally CrateDB could handle this e.g. like DuckDB and map types where possible and otherwise use generic `json` type

```sql
SELECT json_structure('[{"x":1,"y":1},{"x":2,"y":[1,2,3]}]') a;
┌──────────────────────────────┐
│              a               │
│             json             │
├──────────────────────────────┤
│ [{"x":"UBIGINT","y":"JSON"}] │
└──────────────────────────────┘
```

**Comment by amotl** [comment]
Thank you for your excellent suggestions 💯. I followed @proddata's advise:
> I think the option is to store what is possible in a typed field and what is not in a separate column.

- GH-39

Please add your voice about general idea, naming things, or anything else which comes to mind. Thanks!

**Comment by amotl** [comment]
The patch referenced above, GH-39, will be included into the upcoming release.

**Comment by amotl** [comment]
The improvement has been included into release v0.0.14.
